x-pack/plugins/security_solution/public/actions/add_to_timeline/data_provider.ts

```diff
@@ -67,12 +67,16 @@ export const createDataProviders = ({
   const arrayValues = Array.isArray(values) ? values : [values];
   return arrayValues.reduce<DataProvider[]>((dataProviders, value, index) => {
     let id: string = '';
-    const appendedUniqueId = `${contextId}-${eventId}-${field}-${index}-${value}`;
+    const appendedUniqueId = `${contextId}${
+      eventId ? `-${eventId}` : ''
```

Comment:
`eventId` is optional, but it is part of `uniqueId` which would result in an id containing  `...-undefined-...`.

@semd Why do we need `eventId`? It is only being used for building the id. 🤔

Reply:
Hi @machadoum 
Yes, this ID has behavior importance in the "addToTimeline" action: the data provider ignores duplicate IDs in the timeline.
Currently, event tables pass the `eventId` and it is used for building the ID, so we are applying this dedup logic only when the "addToTimeline" action is clicked in the same table cell.

https://user-images.githubusercontent.com/17747913/211304384-e81285f7-b177-4270-9675-2fd63aa2eb97.mov

I am not sure why the dedup is scoped by event, it feels like a bug to me, but probably there was a reason for being implemented. I think we should discuss this behavior with the team, if we decide to apply the dedup in a general way then we'll be able to get rid of `eventId`.